### PR TITLE
Trigger FlexCI from GitHub Actions

### PR DESCRIPTION
**Pull request by kmaehashi**
Trigger FlexCI from GitHub Actions. Benefits are:

* Display tests as `Skipped` rather than `succeeded` for skipped tests.
* Can incrementally run tests (e.g., run `/test mini`, then run `/test array-api`) while preserving all CI results in Checklist
* Do not launch unnecessary FlexCI instance just to skip tests
* Make webhook management easier

### CODE REVIEW

File: .github/workflows/flexci_dispatcher.py [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+            _log(f'Triggered: {base_url}/r/job/{job["id"]}')
+        return True
+    elif 'message' in response:
+        _log(f'Failed to submit webhook payload: {response["message"]}')
+        return False
+    raise RuntimeError('unexpected response: {response}')
+
+
+def _complement_commit_status(
+        repo: str, pull_req: int, token: Optional[str],
```
Comment (toslunar) [inline .github/workflows/flexci_dispatcher.py:60]:
```suggestion
        repo: str, pull_req: int, token: str,
```

File: .github/workflows/flexci_dispatcher.py [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+    parser.add_argument(
+        '--flexci-context', type=str, default='pfn-public-ci',
+        help='Context prefix of the FlexCI server (default: %(default)s)')
+    return parser.parse_args(argv[1:])
+
+
+def main(argv: Any) -> int:
+    options = parse_args(argv)
+    webhook_secret = str(os.environ.get('FLEXCI_WEBHOOK_SECRET'))
+    github_token = str(os.environ.get('GITHUB_TOKEN'))
```
Comment (toslunar) [inline .github/workflows/flexci_dispatcher.py:108]: If these env variables are required,
```suggestion
    webhook_secret = os.environ['FLEXCI_WEBHOOK_SECRET']
    github_token = os.environ['GITHUB_TOKEN']
```

**Comment by kmaehashi** [comment]
Applied review suggestions!

**Review by toslunar — APPROVED** [review]
LGTM

**Pull request merged**